cli/src/pcluster/templates/awsbatch_builder.py

```diff
@@ -247,10 +246,12 @@ def _add_job_role(self):
                                 resources=[
                                     self._format_arn(
                                         service="cloudformation",
-                                        resource="stack/parallelcluster-*/*",
-                                        region=self._stack_region,
-                                        account=self._stack_account,
-                                    )
+                                        resource=f"stack/{self.stack_name}/*",
+                                    ),
+                                    self._format_arn(
+                                        service="cloudformation",
+                                        resource=f"stack/{self.stack_name}-*/*",
```

Comment:
Why do we need this {self.stack_name}-* rule?
I think it was here because of the substack names.

Reply:
If it's for CFN generated resources let's add a comment.

Reply:
It is for query of substack, although we don't have any substack now. I added ToDos to track it.